app/controllers/course/forum/posts_controller.rb

```diff
@@ -19,6 +19,7 @@ def create
 
     if result
       send_created_notification(current_user, current_course_user, @post)
+      auto_answer_action(@post, @topic) unless current_course_user.staff? || rag_settings[:response_workflow] == 'no'
```

Comment:
Prefer explicit `rag_settings[:response_workflow] != 'yes'` since we note that sometimes `rag_settings[:response_workflow]` can be `nil`.

Also, cleaner to put all your guards to within `auto_answer_action` itself, so that the scope of Rag is kept within that file (as far as possible).

Reply:
I have shifted all guards to be within auto_answer_action itself